CMakeLists.txt

```diff
@@ -96,13 +101,13 @@ set_target_properties(lib_kmeans
         PROPERTIES
         VERSION 0.0.1
         SOVERSION 0.0.1
-        LIBRARY_OUTPUT_NAME intmpi
+        LIBRARY_OUTPUT_NAME kmeans
         ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_Fortran_LIB_DIRECTORY}
         LIBRARY_OUTPUT_DIRECTORY ${CMAKE_Fortran_LIB_DIRECTORY}
         )
 
 # Linked libraries
-target_link_libraries(lib_kmeans PRIVATE OpenMP::OpenMP_Fortran ${MPI_Fortran_LIBRARIES} common_options)  # BLAS::BLAS 
+target_link_libraries(lib_kmeans PRIVATE OpenMP::OpenMP_Fortran MPI::MPI_Fortran common_flags common_options)
```

Comment:
```suggestion
target_link_libraries(lib_kmeans PRIVATE
    OpenMP::OpenMP_Fortran
    $<$<BOOL:${MPI}>:MPI::MPI_Fortran>
    common_flags common_options
)
```

Reply:
Ah, is this the correct thing to do? Never seen this before